tools/ami-cleanup/src/cleanup/cleanup_test.go

```diff
@@ -0,0 +1,677 @@
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	ec2types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/gravitational/trace"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/slices"
+)
+
+func TestGetEnabledRegions(t *testing.T) {
+	// These are defined as separate variables so that pointers can reference the values
+	usEast1 := "us-east-1"
+	usEast2 := "us-east-2"
+	usWest1 := "us-west-1"
+	usWest2 := "us-west-2"
+	expectedRegions := []accounttypes.Region{
+		{
+			RegionName:      &usEast1,
+			RegionOptStatus: accounttypes.RegionOptStatusEnabled,
+		},
+		{
+			RegionName:      &usEast2,
+			RegionOptStatus: accounttypes.RegionOptStatusEnabledByDefault,
+		},
+		{
+			RegionName:      &usWest1,
+			RegionOptStatus: accounttypes.RegionOptStatusEnabledByDefault,
+		},
+		{
+			RegionName:      &usWest2,
+			RegionOptStatus: accounttypes.RegionOptStatusEnabled,
+		},
+	}
+
+	tests := []struct {
+		desc            string
+		mockListRegions func(ctx context.Context, params *account.ListRegionsInput, optFns ...func(*account.Options)) (*account.ListRegionsOutput, error)
+		expectedRegions []accounttypes.Region
+		checkError      require.ErrorAssertionFunc
+	}{
+		{
+			desc: "fail if API call errors",
+			mockListRegions: func(ctx context.Context, params *account.ListRegionsInput, optFns ...func(*account.Options)) (*account.ListRegionsOutput, error) {
+				return nil, errors.New("some API call error")
+			},
+			checkError: require.Error,
+		},
+		{
+			desc: "no error when API call does not error",
+			mockListRegions: func(ctx context.Context, params *account.ListRegionsInput, optFns ...func(*account.Options)) (*account.ListRegionsOutput, error) {
+				return &account.ListRegionsOutput{}, nil
+			},
+		},
+		{
+			desc: "only enabled regions requested",
+			mockListRegions: func(ctx context.Context, params *account.ListRegionsInput, optFns ...func(*account.Options)) (*account.ListRegionsOutput, error) {
+				enabledStatuses := []accounttypes.RegionOptStatus{accounttypes.RegionOptStatusEnabled, accounttypes.RegionOptStatusEnabledByDefault}
+				slices.Sort(enabledStatuses)
+				slices.Sort(params.RegionOptStatusContains)
+				if slices.Compare(enabledStatuses, params.RegionOptStatusContains) != 0 {
+					return nil, trace.Errorf("the requested region statuses %#v did not match the expected region statuses %#v", params.RegionOptStatusContains, enabledStatuses)
+				}
+				return &account.ListRegionsOutput{}, nil
+			},
+		},
+		{
+			desc: "all results are returned",
+			mockListRegions: func(ctx context.Context, params *account.ListRegionsInput, optFns ...func(*account.Options)) (*account.ListRegionsOutput, error) {
+				return &account.ListRegionsOutput{
+					Regions:   expectedRegions,
+					NextToken: nil,
+				}, nil
+			},
+			expectedRegions: expectedRegions,
+		},
+	}
+
+	for _, test := range tests {
+		if test.checkError == nil {
+			test.checkError = require.NoError
+		}
```

Comment:
Personally, I'd make sure to set `checkError` in all test cases above and then do away with this if. Less conditionals are easier to reason about and it makes your test cases more explicit.